src/Symfony/Component/Notifier/Bridge/Sweego/SweegoTransport.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Sweego;
+
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Mathieu Santostefano <msantostefano@protonmail.com>
+ */
+final class SweegoTransport extends AbstractTransport
+{
+    protected const HOST = 'api.sweego.io';
+
+    public function __construct(
+        #[\SensitiveParameter] private string $apiKey,
+        private string $region,
+        private string $campaignType,
+        ?HttpClientInterface $client = null,
+        ?EventDispatcherInterface $dispatcher = null,
+    ) {
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return \sprintf('sweego://%s?region=%s&campaign_type=%s', $this->getEndpoint(), $this->region, $this->campaignType);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage && null === $message->getOptions();
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $endpoint = \sprintf('https://%s/send', $this->getEndpoint());
+        $response = $this->client->request('POST', $endpoint, [
+            'headers' => [
+                'Api-Key' => $this->apiKey,
+            ],
+            'json' => [
+                'recipients' => [
```

Comment:
After having the PR reviewed with my CTO, we saw that there was some missing optional fields

1. bat [boolean]: enable test mode (no sms sent, it's for testing purpose)
2. campaign_id [string]: used for tracking / filtering purpose on our platform; identity an SMS campaign and allow to see logs / stats only for this campaign
3. shorten_urls [boolean]: True by default, we replaces all url in the SMS content by a shortened url version (reduce the characters of the sms)
4. shorten_with_protocol [boolean]: True by default, add scheme to shortened url version
5. template_id [UUID4] / variables [Dictionary [ string, string ]] : it could be used in replacement of the message_txt field. (only one of template_id / message_txt accepted). Users can creates templates on our platform that will contains variables placeholders and use the variables field to fill the placeholders with the given variables 

Would it be possible to also add those fields ?

Reply:
templates should not be added here. Using a template is totally different than passing the message text, and does not correspond to a usage of the notifier component (remember that this package is not a SDK for the full Sweego API but a bridge to make it easy to use the Notifier features with Sweego).

`bat` probably makes sense as an option in the DSN. The `shorten_*` settings might also be exposed in the DSN.

For `campaign_id`, I'm not sure what is the best mapping.

Reply:
I'm making those options available through DSN, via an `Options` class (like for Lox24 Bridge). It means those options could be set via the DSN or at runtime by creating the `SweegoOptions` class withe specific values. It will overwrite the existing ones from the DSN. 

I'm implementing: `bat`, `campaignId`, `shortenUrls` and `shortenWithProtocol`.